src/viewport.js

```diff
@@ -27,12 +27,20 @@ export function Viewport(container, observerCollection) {
   const customHandleScrollResize = observerCollection.handleScrollResize
   const handler = (this.handler = customHandleScrollResize ? customHandleScrollResize(_handler) : _handler)
 
-  addEventListener('scroll', handler, true)
-  addEventListener('resize', handler, true)
-  addEventListener('DOMContentLoaded', () => {
+  const mutationObserverCallback = () => {
     const mutationObserver = (this.mutationObserver = new MutationObserver(_handler))
     mutationObserver.observe(document, { attributes: true, childList: true, subtree: true })
-  })
+  }
+
+  addEventListener('scroll', handler, true)
+  addEventListener('resize', handler, true)
+
+  const documentReadyState = document.readyState
+  if (documentReadyState === 'complete' || documentReadyState === 'interactive') {
```

Comment:
`document.readyState !== 'loading'` could be shorter (there are only 3 states)

Reply:
You’re right, changed!